File: src/v2/pyZOMBIA.io/src/server/Network/SendEntityUpdate.py
from Network.PacketIds import packets
from Util.Util import Util

import msgpack
import asyncio
import websockets

def SendEntityUpdate(socket, packet):
  try:
    asyncio.get_event_loop().run_until_complete(socket.send(msgpack.packb({
      "opcode": packets["PACKET_ENTITY_UPDATE"],
      "tick": Util.getTick(),
      "entities": packet
    })))
  except websockets.exceptions.ConnectionClosed:
    try:
      del data["clients"][Util.socketUid(websocket)]
    except Exception:
      pass

File: src/server/Network/SendEntityUpdate.js
import SERVER_DATA from "data/all";
import PacketIds from "Network/PacketIds";
import * as msgpack from "@msgpack/msgpack";

function SendEntityUpdate({
  entities,
  socket
}) {
  if (socket && socket.readyState !== socket.OPEN) return;

  socket.send(msgpack.encode({
    opcode: PacketIds.PACKET_ENTITY_UPDATE,
    tick: SERVER_DATA.TICK,
    entities
  }));
}

export default SendEntityUpdate;
